lib/Mongo/MongoCollection.php

```diff
@@ -183,7 +210,13 @@ public function getSlaveOkay()
      */
     public function setSlaveOkay($ok = true)
     {
-        $this->notImplemented();
+        $result = $this->getSlaveOkay();
+        $this->readPreference = new ReadPreference(
+            $ok ? ReadPreference::RP_SECONDARY : ReadPreference::RP_PRIMARY,
```

Comment:
In the driver, `slaveOkay` sets a readPreference of `RP_SECONDARY_PREFERRED`: https://github.com/mongodb/mongo-php-driver-legacy/blob/master/collection.c#L203